File: 0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.js
/**
 * @param {number[]} cost
 * @return {number}
 */
const minCostClimbingStairs = cost => {
    const newArr = new Array(cost.length);
    if (cost.length === 1) return cost[0];
    if (cost.length === 2) return Math.min(cost[0], cost[1]);
    newArr[cost.length - 1] = cost[cost.length - 1];
    newArr[cost.length - 2] = cost[cost.length - 2];
    for (let i = cost.length - 3; i >= 0; i--) {
        newArr[i] = Math.min(newArr[i + 2], newArr[i + 1]) + cost[i];
    }
    return Math.min(newArr[0], newArr[1]);
};

File: 0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        arr = [None] * len(cost)
        arr[-1] = cost[-1]
        arr[-2] = cost[-2]
        i = len(cost) - 3
        while i >= 0:
            arr[i] = min(cost[i] + arr[i + 1], cost[i] + arr[i + 2])
            i -= 1
        return min(arr[0], arr[1])